rollup.config.js

```diff
@@ -47,37 +40,94 @@ const plugins = [
   commonjs({
     ignoreGlobal: true,
   }),
-  !esbundle && replace({
-    'process.env.NODE_ENV': JSON.stringify(prod ? 'production' : 'development'),
-  }),
-  prod && inject({
-    process: processShim,
-  }),
   babel({
     babelrc: false,
-    presets: [
-      ['env', { modules: false, loose: true }],
-      'react',
-    ],
-    plugins: [
-      !prod && 'flow-react-proptypes',
-      prod && 'transform-react-remove-prop-types',
-      'transform-flow-strip-types',
-      'external-helpers',
-      'transform-object-rest-spread',
-      'transform-class-properties',
-    ].filter(Boolean),
+    ...babelrc,
+    plugins: babelrc.plugins.concat('external-helpers'),
   }),
-].filter(Boolean)
-
-if (prod) plugins.push(uglify(), visualizer({ filename: './bundle-stats.html' }))
+]
 
-export default {
-  entry: 'src/index.js',
-  moduleName: 'styled',
-  external: ['react'].concat(esbundle ? Object.keys(pkg.dependencies) : []),
-  exports: 'named',
-  targets,
-  plugins,
+const configBase = {
+  input: 'src/index.js',
   globals: { react: 'React' },
+  external: ['react'].concat(Object.keys(pkg.dependencies)),
+  plugins: commonPlugins,
+}
+
+const umdConfig = {
+  ...configBase,
+  output: {
+    file: 'dist/styled-components.js',
+    format: 'umd',
+    name: 'styled',
+    exports: 'named',
+  },
+  external: ['react'],
+}
+
+const devUmdConfig = {
+  ...umdConfig,
+  plugins: umdConfig.plugins.concat(
+    replace({
+      'process.env.NODE_ENV': JSON.stringify('development'),
+    }),
+  ),
 }
+
+const prodUmdConfig = {
+  ...umdConfig,
+  output: {
+    ...umdConfig.output,
+    file: 'dist/styled-components.min.js',
+  },
+  plugins: umdConfig.plugins.concat(
+    replace({
+      'process.env.NODE_ENV': JSON.stringify('production'),
+    }),
+    inject({
+      process: processShim,
```

Comment:
Why do we not need the process shim for the dev umd bundle?

Reply:
have no idea why it is here at all for any bundle type - i just have left this untouched, it was already there before my PR

Reply:
Let's try creating a bundle with and without it and look at the diff? If there's none we should just get rid of it. :grin: